assets/test/e2e/test.js

```diff
@@ -33,9 +33,100 @@ describe('', function() {
+    expect(element(by.model('displayName')).isPresent()).toBe(true);
+    expect(element(by.model('description')).isPresent()).toBe(true);
+  }
+
+  var requestCreateFromSource = function(projectName, sourceUrl) {
+    var uri = '/project/' + projectName + '/create';
+    expect(browser.getCurrentUrl()).toContain(uri);
+    element(by.model('from_source_url')).clear().sendKeys(sourceUrl);
+    var nextButton = element(by.buttonText('Next'));
+    browser.wait(protractor.ExpectedConditions.elementToBeClickable(nextButton), 2000);
+    return nextButton.click();
+  }
+
+  var requestCreateFromTemplate = function(projectName, templateName) {
+    var uri = '/project/' + projectName + '/create';
+    expect(browser.getCurrentUrl()).toContain(uri);
+    var template = element(by.cssContainingText('.catalog h3 > a', templateName));
+    expect(template.isPresent()).toBe(true);
+    return template.click();
+  }
+
+  var attachBuilderImageToSource = function(projectName, builderImageName) {
+    var uri = '/project/' + projectName + '/catalog/images';
+    expect(browser.getCurrentUrl()).toContain(uri);
+    expect(element(by.cssContainingText('h1', "Select a builder image")).isPresent()).toBe(true);
+    var builderImageLink = element(by.cssContainingText('div > h3 > a', builderImageName));
+    expect(builderImageLink.isPresent()).toBe(true);
+    return builderImageLink.click();
+  }
+
+  var createFromSourceSummary = function(projectName, builderImageName, appName) {
+    var uri = '/project/' + projectName + '/create/fromimage';
+    expect(browser.getCurrentUrl()).toContain(uri);
+    expect(element(by.css('.create-from-image h1')).getText()).toEqual(builderImageName);
+    expect(element(by.cssContainingText('div > h2', "Name")).isPresent()).toBe(true);
+    expect(element(by.cssContainingText('div > h2', "Routing")).isPresent()).toBe(true);
+    expect(element(by.cssContainingText('div > h2', "Deployment Configuration")).isPresent()).toBe(true);
+    expect(element(by.cssContainingText('div > h2', "Build Configuration")).isPresent()).toBe(true);
+    expect(element(by.cssContainingText('div > h2', "Scaling")).isPresent()).toBe(true);
+    expect(element(by.cssContainingText('div > h2', "Labels")).isPresent()).toBe(true);
+    element(by.name('appname')).clear().sendKeys(appName);
+    return element(by.buttonText("Create")).click();
+  }
+
+  var createFromTemplateSummary = function(projectName, templateName, parameterNames, labelNames) {
+    var uri = '/project/' + projectName + '/create/fromtemplate';
+    expect(browser.getCurrentUrl()).toContain(uri);
+    expect(element(by.css('.create-from-template h1')).getText()).toEqual(templateName);
+    expect(element(by.cssContainingText('div > h2', "Images")).isPresent()).toBe(true);
+    expect(element(by.cssContainingText('div > h2', "Parameters")).isPresent()).toBe(true);
+    expect(element(by.cssContainingText('div > h2', "Labels")).isPresent()).toBe(true);
+    if (parameterNames) {
```

Comment:
I considered that, but decided to keep defaults for now. That would definitely be a nice improvement (not sure if in the scope of integration, but still nice).